File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/PECMails/PECMailLog.cs
using System;
using VecompSoftware.DocSuiteWeb.Repository.Entity;

namespace VecompSoftware.DocSuiteWeb.Entity.PECMails
{
    public class PECMailLog : DSWBaseEntity, IUnauditableEntity
    {
        #region [ Constructor ]

        public PECMailLog() : this(Guid.NewGuid()) { }

        public PECMailLog(Guid uniqueId)
            : base(uniqueId)
        {
        }

        #endregion

        #region [ Properties ]        

        public string Description { get; set; }

        public string LogType { get; set; }

        public DateTime LogDate { get; set; }

        public string SystemComputer { get; set; }

        public short? Severity { get; set; }

        public string Hash { get; set; }

        #endregion

        #region [ Navigation Properties ]

        public PECMail PECMail { get; set; }

        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/PEC/PECMailLog.vb

<Serializable()>
Public Class PECMailLog
    Inherits DomainObject(Of Int32)

#Region " Fields "

#End Region

#Region " Constructor "

    Public Sub New()
        UniqueId = Guid.NewGuid()
    End Sub

#End Region

#Region " Properties "

    Public Overridable Property Mail As PECMail

    Public Overridable Property Description As String

    Public Overridable Property Type As String

    Public Overridable Property [Date] As DateTime

    Public Overridable Property SystemComputer As String

    Public Overridable Property SystemUser As String

    Public Overridable Property DestinationMail As PECMail

    Public Overridable Property Severity As Short?

    Public Overridable Property Hash As String


#End Region

#Region " Methods "

    Public Overridable Function SetLogType(logType As PECMailLogType) As PECMailLog
        Me.Type = logType.ToString()
        Return Me
    End Function

#End Region

End Class
